dartagnan/src/main/java/com/dat3m/dartagnan/Dartagnan.java

```diff
@@ -54,6 +59,19 @@ public static void main(String[] args) throws Exception {
 
         WitnessGraph witness = new WitnessGraph();
         
+        ShutdownManager sdm = ShutdownManager.create();
+    	Thread t = new Thread(() -> {
+			try {
+				if(options.getSettings().getSolverTimeout() > 0) {
+					// Converts timeout from secs to millisecs
+					Thread.sleep(1000 * options.getSettings().getSolverTimeout());
+					sdm.requestShutdown("Shutdown Request");
+					logger.warn("Shutdown Request");
+				}
+			} catch (InterruptedException e) {
+				// Verification ended, nothing to be done.
+			}});
+        
```

Comment:
The Thread creation should be moved more closely to the place of usage. Also, creating it only if a timeout is specified might be more efficient.

Reply:
As discussed, the performance overhead is almost null so no need to create it only if a timeout is passed. I'm not sure what you mean by `closely to the place of usage` ... the timeout is supposed to be used for the whole running of the process so I place it early in the running process

Reply:
I'm talking about the fact that the created Thread is started over 40 lines later in the code, so its creation can be moved to the point of first usage (around line 107).

Reply:
Actually, my comment above contradicts the the decision I made when I put the `start` at that point. I'm moving all those together at the point where the start currently is